### PR TITLE
Speed up plan selector by reducing Cost API calls

### PR DESCRIPTION
**Pull request by dangodev**
## Reason for change

`<manifold-plan-selector>` currently uses the cost API a little too much. This **🚀 delivers a UI perf boost** by reducing cost API calls to only be used for custom plans. Otherwise it uses built-in plan data. In addition to improving the perceived load time, it will also dramatically reduce the load on our cost API

This also fixes all the mysterious errors happening when plans are changed in `<manifold-plan-selector>` (well, except for the JawsDB custom plan, that is).

Closes manifoldco/engineering#8619
Closes manifoldco/engineering#8209

## Testing
Click through the JawsDB plan, and confirm the prices are correct. Also make sure the custom plan prices are correct as well.

### CODE REVIEW

File: src/utils/plan.ts [hunk 1]
```diff
@@ -272,6 +272,20 @@ export function initialFeatures(features: Catalog.ExpandedFeature[]): Gateway.Fe
...
+  return [...plans].sort((a, b) => {
+    // Free plans up front
+    if (a.body.free === true || b.body.free === true) {
+      return a.body.free === true ? -1 : 1;
+    }
+    // Sort by cost
+    return (a.body.defaultCost || a.body.cost) - (b.body.defaultCost || b.body.cost);
+  });
+}
+
```
Comment (dangodev) [inline src/utils/plan.ts]: Added plan sort function + test; the logic was getting to be a little much within the top-level component.

Comment (davidleger95) [reply src/utils/plan.ts]: do custom plans go to the bottom?

Comment (dangodev) [reply src/utils/plan.ts]: That is a real good question. I am not sure either TBH

Comment (dangodev) [reply src/utils/plan.ts]: Oh there’s another sort function happening in `manifold-plan-menu`. Good catch! Let’s combine those.

File: src/components/manifold-plan-menu/manifold-plan-menu.tsx [hunk 2]
```diff
@@ -33,25 +33,15 @@ export class ManifoldPlanMenu {
...
   @Prop() selectPlan: Function = () => {};
 
-  sortPlans(plans: Catalog.ExpandedPlan[]) {
-    return [...plans].sort((a, b) => {
-      const aIndex = a.body.customizable ? plans.length : 0;
-      const bIndex = b.body.customizable ? plans.length : 0;
-      return aIndex - bIndex;
-    });
-  }
-
```
Comment (dangodev) [inline src/components/manifold-plan-menu/manifold-plan-menu.tsx]: Combined into `planSort()` method

File: src/components/manifold-plan-menu/manifold-plan-menu.spec.ts [hunk 3]
```diff
@@ -1,17 +0,0 @@
...
-    const nonCustom = { ...ExpandedPlan, body: { ...ExpandedPlan.body, customizable: undefined } };
-    const custom = { ...ExpandedPlan, body: { ...ExpandedPlan.body, customizable: true } };
-    expect(planMenu.sortPlans([nonCustom, custom, nonCustom, custom])).toEqual([
-      nonCustom,
-      nonCustom,
-      custom,
-      custom,
-    ]);
-  });
-});
```
Comment (dangodev) [inline src/components/manifold-plan-menu/manifold-plan-menu.spec.ts]: Now we’re testing custom plan sorting (and more!) in `plan.spec.ts` so this test was just deleted

**Comment by dangodev** [comment]
Going to hold off merging this until the preview links work again.

**Pull request merged**